collector/container/rhtap-slim.Dockerfile

```diff
@@ -0,0 +1,151 @@
+ARG CMAKE_BUILD_TYPE=Release
+ENV CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+
+# Appends an argument to the driver download URL that is used for filtering alerts on missing kernels.
+# TODO(ROX-20240): This needs to be true on PRs only.
+ARG COLLECTOR_APPEND_CID=false
+ENV COLLECTOR_APPEND_CID=${COLLECTOR_APPEND_CID}
+
+WORKDIR ${BUILD_DIR}
+
+RUN mkdir kernel-modules \
+    && cp -a /builder builder \
+    && cp -a /collector collector \
+    && cp -a /falcosecurity-libs falcosecurity-libs \
+    && cp -a /builder/third_party third_party \
+    && cp -a /kernel-modules/MODULE_VERSION kernel-modules/MODULE_VERSION \
+    && cp -a /CMakeLists.txt CMakeLists.txt
+
+# WITH_RHEL_RPMS controls for dependency installation, ie if they were already installed as RPMs.
+# Setting the value to empty will cause dependencies to be downloaded from repositories or accessed in submodules and compiled.
+# TODO(ROX-20651): Set ENV WITH_RHEL_RPMS=true when RHEL RPMs can be installed to enable hermetic builds.
+RUN ./builder/install/install-dependencies.sh \
+    && cmake -DDISABLE_PROFILING=ON -S "${SRC_ROOT_DIR}" -B "${CMAKE_BUILD_DIR}" \
+    && cmake --build "${CMAKE_BUILD_DIR}" --target all -- -j "$(nproc)" \
+    && (cd ${CMAKE_BUILD_DIR} && ctest -V) \
+    && strip --strip-unneeded "${CMAKE_BUILD_DIR}/collector/collector" "${CMAKE_BUILD_DIR}/collector/EXCLUDE_FROM_DEFAULT_BUILD/libsinsp/libsinsp-wrapper.so"
+
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.2
+
+ARG BUILD_TYPE=rhel
+ARG ROOT_DIR=.
+# TODO(ROX-20236): configure injection of dynamic version value when it becomes possible.
+ARG COLLECTOR_VERSION=0.0.1-todo
+ARG MODULE_VERSION=0.0.1-todo
+
+ENV ROOT_DIR=$ROOT_DIR
+ENV COLLECTOR_VERSION="${COLLECTOR_VERSION}"
+
+ENV COLLECTOR_HOST_ROOT=/host
+
+LABEL \
+    com.redhat.component="rhacs-collector-slim-container" \
+    com.redhat.license_terms="https://www.redhat.com/agreements" \
+    description="This image supports runtime data collection in the StackRox Kubernetes Security Platform" \
+    io.k8s.description="This image supports runtime data collection in the StackRox Kubernetes Security Platform" \
+    io.k8s.display-name="collector-slim" \
+    io.openshift.tags="rhacs,collector,stackrox" \
+    maintainer="Red Hat, Inc." \
+    name="rhacs-collector-slim-rhel8" \
+    source-location="https://github.com/stackrox/collector" \
+    summary="Runtime data collection for the StackRox Kubernetes Security Platform" \
+    url="https://catalog.redhat.com/software/container-stacks/detail/60eefc88ee05ae7c5b8f041c" \
+    version=${COLLECTOR_VERSION} \
+    collector_version=${COLLECTOR_VERSION}
+    # TODO: These two labels only exist on upstream, keep or remove?
+    # io.stackrox.collector.module-version="${MODULE_VERSION}" \
+    # io.stackrox.collector.version="${COLLECTOR_VERSION}"
+
+WORKDIR /
+
+COPY collector/container/${BUILD_TYPE}/install.sh /
+RUN ./install.sh && rm -f install.sh
```

Comment:
Let's please use this as an opportunity to also improve this `install.sh`:

Doing `microdnf -y upgrade --nobest` instead of just `microdnf -y upgrade` saves us from occasional failures when RPM repo isn't in a consistent state.